### PR TITLE
`:symbol` shorthand for `Symbol.symbol` or `Symbol.for("symbol")`

### PR DESCRIPTION
**Pull request by edemaine**

### CODE REVIEW

File: source/parser.hera [hunk 1]
```diff
@@ -1580,7 +1597,8 @@ PropertyAccessModifier
 PropertyAccess
   # NOTE: Added shorthand x."string" -> x["string"]
   # NOTE: Added shorthand x.3 -> x[3]
-  AccessStart:dot ( TemplateLiteral / StringLiteral / IntegerLiteral ):literal ->
+  # NOTE: Added shorthand x.:symbol -> x[Symbol.for("symbol")]
```
Comment (STRd6) [inline source/parser.hera:1600]: I hand't thought about this but it's a nice and convenient shorthand!

**Review by STRd6 — APPROVED** [review]
Looks great! Glad it was so simple.

P.S. did you `git add` your tests? :P

File: source/parser.hera (hunk 1 above)
Comment (edemaine) [reply source/parser.hera:1600]: Thanks! I looked at all the places we had `LengthShorthand` already, and extended those in some natural ways. `x.:symbol` looks a little funny but `x:symbol` looks quite nice (we support both). Overall, I'm excited about how this makes well-known symbols so much easier to use, especially to make custom iterators and such.

Comment (edemaine) [reply source/parser.hera:1600]: Whoops, `x:symbol` conflicts with object literals, so we can't use that. But `x.:symbol` remains.

Comment (STRd6) [reply source/parser.hera:1600]: I'd be ok with forcing inline object literals to have whitespace but we probably want to do a brief survey of existing code to make sure it is very rare or non-existent to write them that way.

Comment (STRd6) [reply source/parser.hera:1600]: Sounds good, we can save it for Civet 2: The Wrath of Khan

**Pull request merged**